src/validationinterface.h

```diff
@@ -52,6 +53,8 @@ struct CMainSignals {
     boost::signals2::signal<void (const CBlockIndex *, const CBlockIndex *, bool fInitialDownload)> UpdatedBlockTip;
     /** Notifies listeners of a transaction having been added to mempool. */
     boost::signals2::signal<void (const CTransactionRef &)> TransactionAddedToMempool;
+    /** Notifies listeners of a transaction having been added to the wallet. */
+    boost::signals2::signal<void (const CTransactionRef &, uint256 hashBlock)> TransactionAddedWallet;
```

Comment:
Throughout the PR, should use `const uint256&` instead of `uint256` arg type to avoid unnecessary copies.